src/transactions/test/AllowTrustTests.cpp

```diff
@@ -690,6 +695,44 @@ template <int V> struct TestStub
                 });
             }
         }
+
+        SECTION("with clawback")
+        {
+            for_versions_from(17, *app, [&] {
+                auto toSet = static_cast<uint32_t>(AUTH_CLAWBACK_ENABLED_FLAG |
+                                                   AUTH_REVOCABLE_FLAG);
+                gateway.setOptions(setFlags(toSet));
+                a1.changeTrust(idr, trustLineLimit);
+
+                SECTION(
+                    "remove offers by pulling auth while clawback is enabled")
+                {
+                    auto market = TestMarket{*app};
+                    auto native = makeNativeAsset();
+
+                    auto offer = market.requireChangesWithOffer({}, [&] {
+                        return market.addOffer(a1,
+                                               {native, idr, Price{1, 1}, 1});
+                    });
+
+                    market.requireChanges(
+                        {{offer.key, OfferState::DELETED}},
+                        [&] { gateway.denyTrust(idr, a1, flagOp); });
+
+                    REQUIRE(
+                        isClawbackEnabledOnTrustline(a1.loadTrustLine(idr)));
+                }
+
+                gateway.allowMaintainLiabilities(idr, a1, flagOp);
```

Comment:
It might be clearer what you are trying to do here if you put this all in a section too. Right now, I'm not sure if it is going to run once or twice.

Reply:
Good point. Fixed.